dbt/adapters/glue/impl.py

```diff
@@ -620,9 +662,135 @@ def hudi_merge_table(self, target_relation, request, primary_key, partition_key,
+                        SELECT * FROM tmp_{target_relation.name}
+                    """
+        else : 
+            query = f"""
+                        INSERT INTO glue_catalog.{target_relation.schema}.{target_relation.name}
+                        SELECT * FROM tmp_{target_relation.name} ORDER BY {partition_by}
+                    """
+        return query
+        
+
+    def iceberg_create_table(self, target_relation, partition_by, location, table_properties):
+        table_properties = self.set_table_properties(table_properties)
+        if partition_by is None: 
+            query = f"""
+                        CREATE TABLE glue_catalog.{target_relation.schema}.{target_relation.name}
+                        USING iceberg
+                        LOCATION '{location}'
+                        {table_properties}
+                        AS SELECT * FROM tmp_{target_relation.name}
+                    """
+        else : 
+            query = f"""
+                        CREATE TABLE glue_catalog.{target_relation.schema}.{target_relation.name}
+                        USING iceberg
+                        PARTITIONED BY {partition_by}
+                        LOCATION '{location}'
+                        {table_properties}
+                        AS SELECT * FROM tmp_{target_relation.name} ORDER BY {partition_by}
+                    """
+        return query
+    
+    
+    def iceberg_upsert(self, target_relation, merge_key):
+    ## Perform merge operation on incremental input data with MERGE INTO. This section of the code uses Spark SQL to showcase the expressive SQL approach of Iceberg to perform a Merge operation
+        query = f"""
+        MERGE INTO glue_catalog.{target_relation.schema}.{target_relation.name} t
+        USING (SELECT * FROM tmp_{target_relation.name}) s
+        ON {self.set_iceberg_merge_key(merge_key=merge_key)}
+        WHEN MATCHED THEN UPDATE SET *
+        WHEN NOT MATCHED THEN INSERT *
+        """
+        return query
+    
+    @available
+    def iceberg_write(self, target_relation, request, primary_key, partition_key, custom_location, write_mode, table_properties):
+        session, client, cursor = self.get_connection()
+        if partition_key is not None:
+            partition_key  = '(' + ','.join(partition_key) + ')'
+        if custom_location == "empty":
+            location = f"{session.credentials.location}/{target_relation.schema}/{target_relation.name}"
+        else:
+            location = custom_location
+        isTableExists = False
+        if self.check_relation_exists(target_relation):
+            isTableExists = True
+        else:
+            isTableExists = False
+        head_code = f'''
+custom_glue_code_for_dbt_adapter
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import *
+warehouse_path = f"{session.credentials.location}/{target_relation.schema}"
+dynamodb_table = 'myGlueLockTable'
```

Comment:
Why this value is hardcoded?

Reply:
I thought it was not useful to let the user decide the table name as it's automatically created. I'll make it customizable in the next PR.